test/base/SynchronizedTest.ooc

```diff
@@ -1,39 +1,39 @@
 use ooc-base
 import threading/Thread, os/Time
 
-TestClass: class {
-	refCounter: ReferenceCounter
-	init: func {
-		refCounter = ReferenceCounter new(this)
+version(debugTests) {
+	TestClass: class {
+		refCounter: ReferenceCounter
+		init: func {
+			refCounter = ReferenceCounter new(this)
+		}
+		dispose: func {
+			refCounter decrease()
+		}
 	}
-	dispose: func {
-		refCounter decrease()
+	
+	obj := TestClass new()
+	
+	thread1 := Thread new(func1)
+	thread2 := Thread new(func2)
+	
+	thread1 start()
+	thread2 start()
+	
+	func1: func {
+		for (i in 0 .. 10) {
+			obj1 := obj
+			Time sleepMilli(100)
+		}
 	}
-}
-
-obj := TestClass new()
-
-thread1 := Thread new(func1)
-thread2 := Thread new(func2)
-
-thread1 start()
-thread2 start()
-
-func1: func {
-	for (i in 0 .. 10) {
-		obj1 := obj
-		Time sleepMilli(100)
+	
+	func2: func {
+		for (i in 0 .. 10) {
+			obj2 := obj
+			Time sleepMilli(76)
+		}
```

Comment:
And isn't it a little strange to define these functions after they were used? Consider moving these lines up to before the other things happen.